programs/cabal-gild.nix

```diff
@@ -0,0 +1,43 @@
+{
+  mkFormatterModule,
+  lib,
+  config,
+  pkgs,
+  ...
+}:
+
+let
+  cfg = config.programs.cabal-gild;
+in
+{
+  meta.maintainers = [ ];
+
+  imports = [
+    (mkFormatterModule {
+      name = "cabal-gild";
+      package = [
+        "haskellPackages"
+        "cabal-gild"
+      ];
+      includes = [
+        "*.cabal"
+        "*cabal.project"
+        "*cabal.project.local"
+      ];
+    })
+  ];
+
+  config = lib.mkIf cfg.enable {
+    settings.formatter.cabal-gild = {
+      # cabal-gild doesn't support multiple file targets, see upstream issue 35
+      command = pkgs.writeShellApplication {
+        name = "cabal-gild-wrapper";
+        text = ''
+          for file in "$@"; do
+            ${lib.getExe cfg.package} --io="$file"
+          done
+        '';
```

Comment:
`args` / `options` is not supported.  Should we fix that?  How?

Reply:
I'm not sure I understand the question. I don't think tools in treefmt-nix *have* to have args or options.

Reply:
If users set `programs.cabal-gild.args = ...`, probably they expect those args to take effect.  In the current implementation, those args are ignored.

Do you mean ignoring args set by users is OK?

Reply:
Ohhhh sorry now I understand.

Is this a problem with all the `for file in "$@"` wrappers we have in treefmt-nix?

Reply:
Yes.  `$@` contains `args`/`options`.  An example value of `$@` can be `arg1 arg2 file1 file2`.

Reply:
What you've done here looks the same as `buf.nix`, `statix.nix`, and others. I think it's fine.